beacon-chain/node/registration/powchain.go

```diff
@@ -0,0 +1,45 @@
+package registration
+
+import (
+	"errors"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/prysmaticlabs/prysm/cmd/beacon-chain/flags"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+// PowchainPreregistration prepares data for powchain.Service's registration.
+func PowchainPreregistration(cliCtx *cli.Context) (depositContractAddress string, endpoints []string, err error) {
+	depositContractAddress, err = DepositContractAddress(cliCtx)
+	if err != nil {
+		return "", nil, err
+	}
+	endpoints = []string{cliCtx.String(flags.HTTPWeb3ProviderFlag.Name)}
+	endpoints = append(endpoints, cliCtx.StringSlice(flags.FallbackWeb3ProviderFlag.Name)...)
+	return
+}
+
+// DepositContractAddress returns the address of the deposit contract.
+func DepositContractAddress(cliCtx *cli.Context) (string, error) {
+	address := params.BeaconConfig().DepositContractAddress
+	if address == "" {
+		return "", errors.New("valid deposit contract is required")
+	}
+
+	if !common.IsHexAddress(address) {
+		return "", errors.New("invalid deposit contract address given: " + address)
+	}
+
+	if cliCtx.String(flags.HTTPWeb3ProviderFlag.Name) == "" {
```

Comment:
This seems like the wrong place to log it, if anything it should be in the above method.

Reply:
You're absolutely right, this mistake must have slipped through when merging manually. Thanks for noticing.